CriteoPublisherSdk/Sources/CRRewardedAdUnit.m

```diff
@@ -29,7 +29,7 @@ - (instancetype)initWithAdUnitId:(NSString *)adUnitId {
 }
 
 - (NSUInteger)hash {
-  return super.hash ^ (NSUInteger)14559042078869117629ull;
+  return super.hash ^ (NSUInteger)14559065388869353629ull;
```

Comment:
Just for my understanding: why and how did you get this value (prime number ?) ? I see that `self.class` is part of the equality, but the hash can be surjective, so why bothering with such number. Moreover, the parent class already contains the ad unit type which kind of of represents the class.

Reply:
I just applied the same logic than the one was in place with other Ad unit types, without thinking that much I admit.